clang/test/SemaObjC/unguarded-availability.m

```diff
@@ -177,16 +177,19 @@ void justAtAvailable(void) {
 
 #ifdef OBJCPP
 
-int f(char) AVAILABLE_10_12;
+int f(char) AVAILABLE_10_12; // #f_char_def
 int f(int);
 
 template <class T> int use_f() {
-  // FIXME: We should warn here!
-  return f(T());
```

Comment:
Could you check that we don't emit a warning if there is an availability attribute on the enclosing function or if the use is guarded by an `if (@available(...))` check? IIRC that that was what I was concerned about when I wrote this. @jansvoboda11 : someone at Apple should probably review this.

Reply:
Thanks for the quick response! I have added the test cases you requested. No additional warnings are emitted.